### PR TITLE
Added PackageId properties

### PR DESCRIPTION
**Pull request by MartinZikmund**
GitHub Issue (If applicable): #1207 

## PR Type

What kind of change does this PR introduce?

- Feature

## What is the current behavior?

`PackageId` returns default values.

## What is the new behavior?

Implemented some of the `PackageId` properties for iOS and Android.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tested code with current [supported SDKs](https://github.com/unoplatform/uno/blob/master/README.md#uno-features)
- [x] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [x] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] [Wasm UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md#running-the-webassembly-ui-tests-snapshots) are not showing unexpected any differences. Validate PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Updated the [Release Notes](https://github.com/unoplatform/uno/tree/master/doc/ReleaseNotes)
- [x] Associated with an issue (GitHub or internal)

### CODE REVIEW

**Comment by MartinZikmund** [comment]
@jeromelaban I'm getting `Error CS0103: The name 'PackageInfoCompat' does not exist in the current context` during the build even though locally it works as expected, what could that be please?

**Comment by jeromelaban** [comment]
It's most probably related to this:
https://github.com/unoplatform/uno/blob/3691ef356b24a369ca334464e109235c0490083a/src/Uno.UWP/Uno.csproj#L5

The CI builds monoandroid80, you should exclude it there.

File: src/Uno.UWP/ApplicationModel/PackageId.iOS.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+		public string Name => NSBundle.MainBundle.InfoDictionary[BundleIdentifierKey].ToString();
+
+		/// <summary>
+		/// Implementation based on <see cref="https://developer.apple.com/documentation/bundleresources/information_property_list/cfbundleversion"/>.
+		/// </summary>
+		public PackageVersion Version
+		{
+			get
+			{
+				var shortVersion = NSBundle.MainBundle.InfoDictionary[BundleShortVersionKey].ToString();
```
Comment (jeromelaban) [inline src/Uno.UWP/ApplicationModel/PackageId.iOS.cs]:
```suggestion
				var shortVersion = NSBundle.MainBundle.InfoDictionary[BundleShortVersionKey].ToString() ?? "0.0";
```

Same for the one below. If for some reason the key does not exists.

File: src/Uno.UWP/ApplicationModel/PackageId.iOS.cs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+				var bundleVersion = NSBundle.MainBundle.InfoDictionary[BundleVersionKey].ToString();
+				// Short version is the user-displayed version, use if possible
+				if (SystemVersion.TryParse(shortVersion, out var userVersoin))
```
Comment (jeromelaban) [inline src/Uno.UWP/ApplicationModel/PackageId.iOS.cs]:
```suggestion
				if (SystemVersion.TryParse(shortVersion, out var userVersion))
```

File: src/Uno.UWP/ApplicationModel/PackageId.iOS.cs [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+namespace Windows.ApplicationModel
+{
+	public partial class PackageId
+	{
+		private const string BundleIdentifierKey = "CFBundleIdentifier";
+		private const string BundleShortVersionKey = "CFBundleShortVersionString";
+		private const string BundleVersionKey = "CFBundleVersion";
+
+		public string FamilyName => NSBundle.MainBundle.InfoDictionary[BundleIdentifierKey].ToString();
```
Comment (jeromelaban) [inline src/Uno.UWP/ApplicationModel/PackageId.iOS.cs]: Check for null

**Comment by MartinZikmund** [comment]
Also fixed the failing iOS Unit Tests for `Launcher` (it was not called from the UI thread which is necessary)

**Comment by jeromelaban** [comment]
I'm adding some more validation through https://github.com/unoplatform/uno/pull/1975, let`s wait a bit for that one.

**Comment by jeromelaban** [comment]
Let's retry with UI tests enabled 🚀

**Pull request merged**